processing/src/main/java/org/apache/druid/segment/nested/StructuredData.java

```diff
@@ -46,7 +47,7 @@ public class StructuredData implements Comparable<StructuredData>
   private static long computeHash(StructuredData data)
   {
     try {
-      final byte[] bytes = ColumnSerializerUtils.SMILE_MAPPER.writeValueAsBytes(data.value);
+      final byte[] bytes = ColumnSerializerUtils.SMILE_MAPPER.writer(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS).writeValueAsBytes(data.value);
```

Comment:
Could you define this as a static final constant `ObjectWriter` in this class so it can be reused if needed instead of instantiating a new object? Also, please include a comment on why this option is needed.